synology_dsm/api/core/share.py

```diff
@@ -66,3 +66,12 @@ def share_size(self, name, human_readable=False):
         if human_readable:
             return SynoFormatHelper.bytes_to_readable(share_size_bytes)
         return share_size_bytes
+
+    def share_attribute(self, name, attribute):
+        """Returns the value of the specified share attribute."""
+        # Makes it easier to get a specific attribute without requiring a
+        # function for each.
+        if attribute in self.share(name):
+            return self.share(name).get(attribute)
+        else:
+            raise ValueError("Specified attribute does not exist: " + attribute)
```

Comment:
I don't think this is a nice function to have because error should be handled by the app that uses the library + it searches every time the right share dict. It's like doing the Python job here. 

In your app you should save the share dict into a variable then use `.get(attr)`.

I don't think this code reduce the amount of code in your app.

But if you want, as we done for `SynoCamera`, you can make a model for the dict.

Reply:
No strong feelings on this one, removed.